legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py

```diff
@@ -596,13 +606,22 @@ def save_filing(  # pylint: disable=too-many-return-statements,too-many-branches
                 return None, None, {"message": f"{business_identifier} not found"}, HTTPStatus.NOT_FOUND
 
             if client_request.method == "PUT":
-                rv = (
-                    db.session.query(LegalEntity, Filing)
-                    .filter(LegalEntity.id == Filing.legal_entity_id)
-                    .filter(LegalEntity.identifier == business_identifier)
-                    .filter(Filing.id == filing_id)
-                    .one_or_none()
-                )
+                if business.is_alternate_name_entity:
+                    rv = (
+                        db.session.query(AlternateName, Filing)
+                        .filter(AlternateName.id == Filing.alternate_name_id)
+                        .filter(AlternateName.identifier == business_identifier)
+                        .filter(Filing.id == filing_id)
+                        .one_or_none()
+                    )
+                else:
+                    rv = (
+                        db.session.query(LegalEntity, Filing)
+                        .filter(LegalEntity.id == Filing.legal_entity_id)
+                        .filter(LegalEntity.identifier == business_identifier)
+                        .filter(Filing.id == filing_id)
+                        .one_or_none()
+                    )
```

Comment:
Think we can simplify and make it one query.

Reply:
Updated, and verified!
![202 put](https://github.com/bcgov/lear/assets/122301442/9aac07c8-c7b2-4290-a631-239263d18688)

Let me know if you're OK with the variable names. My brain went blank on what to name them 😆